storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriteHandler.java

```diff
@@ -130,15 +130,17 @@ public synchronized void write(Collection<ShufflePartitionedBlock> shuffleBlocks
     String dataFileName = ShuffleStorageUtils.generateDataFileName(fileNamePrefix);
     String indexFileName = ShuffleStorageUtils.generateIndexFileName(fileNamePrefix);
 
-    try (LocalFileWriter dataWriter = createWriter(dataFileName, dataBufferSize);
-        LocalFileWriter indexWriter = createWriter(indexFileName, indexBufferSize); ) {
+    try (FileWriter dataWriter =
+            LocalFileWriterFactory.getLocalFileWriter(
+                rssBaseConf, new File(basePath, dataFileName), dataBufferSize);
+        FileWriter indexWriter = createWriter(indexFileName, indexBufferSize)) {
 
       long startTime = System.currentTimeMillis();
       for (ShufflePartitionedBlock block : shuffleBlocks) {
         long blockId = block.getBlockId();
         long crc = block.getCrc();
         long startOffset = dataWriter.nextOffset();
```

Comment:
`dataWriter` could be `null` because you return `null` in the code.

Reply:
Indeed